packages/core/contracts/chainbridge/SinkGovernor.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "../oracle/interfaces/FinderInterface.sol";
+import "../oracle/implementation/Constants.sol";
+
+/**
+ * @title Governor contract on L2 that receives governance actions from L1.
+ */
+contract SinkGovernor {
+    FinderInterface public finder;
+
+    event ExecutedGovernanceTransaction(address indexed to, uint256 value, bytes indexed data);
+
+    constructor(FinderInterface _finder) {
+        finder = _finder;
+    }
+
+    /**
+     * @notice This method will ultimately be called after a governance execution has been bridged cross-chain from Mainnet
+     * to this network via an off-chain relayer. The relayer will call `Bridge.executeProposal` on this local network,
+     * which call `GenericHandler.executeProposal()` and ultimately this method.
+     * @dev This method should send the arbitrary transaction emitted by the L1 governor on this chain.
+     */
+    function executeGovernance(
+        address to,
+        uint256 value,
+        bytes memory data
+    ) external {
+        require(msg.sender == finder.getImplementationAddress(OracleInterfaces.GenericHandler));
```

Comment:
why not add an error description to the require?

Reply:
I can. It seems like we had recently become more comfortable trading require messages for gas (and bytecode) efficiency, but I can add back for better visibility.

Reply:
Done.